modules/project/service_accounts.tf

```diff
@@ -54,10 +58,10 @@ data "google_bigquery_default_service_account" "bq_sa" {
 }
 
 # Secret Manager SA created just in time, we need to trigger the creation.
-resource "google_project_service_identity" "sm_sa" {
+resource "google_project_service_identity" "jit_si" {
+  for_each   = { for service in var.services : service => service if contains(local.jit_services, service) }
```

Comment:
what about
``` setintersection(var.services, local.jit_services)```

IMO that's s little easier to read

Reply:
nice!